src/handler/capabilities.rs

```diff
@@ -0,0 +1,27 @@
+use tower_lsp::lsp_types::{
+    HoverOptions, HoverProviderCapability, SemanticTokensFullOptions, SemanticTokensOptions,
+    SemanticTokensServerCapabilities, ServerCapabilities, TextDocumentSyncCapability,
+    TextDocumentSyncKind, WorkDoneProgressOptions,
+};
+
+use super::wit;
+
+pub fn server_capabilities() -> ServerCapabilities {
+    ServerCapabilities {
+        text_document_sync: Some(TextDocumentSyncCapability::Kind(TextDocumentSyncKind::FULL)),
+        hover_provider: Some(HoverProviderCapability::Options(HoverOptions {
+            work_done_progress_options: WorkDoneProgressOptions {
+                work_done_progress: Some(false),
+            },
+        })),
+        semantic_tokens_provider: Some(SemanticTokensServerCapabilities::SemanticTokensOptions(
+            SemanticTokensOptions {
+                work_done_progress_options: Default::default(),
+                legend: wit::token::legend(),
+                full: Some(SemanticTokensFullOptions::Delta { delta: Some(false) }),
+                range: None,
+            },
+        )),
```

Comment:
by adding the semantic tokens provider here, the TextMate scopes are being overwritten. consequently, the highlighting is different before and after the LSP kicks in. while keeping the aesthetic as close as possible to the current setup, the scopes within `syntaxes/wit.tmLanguage.json` have to be modified to match [this expected mapping table](https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#predefined-textmate-scope-mappings) or we have to define a custom mapping.